src/webgl/p5.Geometry.js

```diff
@@ -174,6 +174,46 @@ p5.Geometry = class Geometry {
     this.vertexColors = [];
     return this;
   }
+
+
+  /**
+ *
+ * @method geometryToObj
+ * @for p5.Geometry
+ */
+
+  geometryToObj (fileName='model.obj'){
+    let objStr= '';
+
+
+    // Vertices
+    this.vertices.forEach(v => {
+      objStr += `v ${v.x} ${v.y} ${v.z}\n`;
+    });
+
+    // Texture Coordinates (UVs)
+    if (this.uvs && this.uvs.length > 0) {
+      for (let i = 0; i < this.uvs.length; i += 2) {
+        objStr += `vt ${this.uvs[i]} ${this.uvs[i + 1]}\n`;
+      }
+    }
+
+    // Vertex Normals
+    if (this.vertexNormals && this.vertexNormals.length > 0) {
+      this.vertexNormals.forEach(n => {
+        objStr += `vn ${n.x} ${n.y} ${n.z}\n`;
+      });
+
+    }
+    // Faces
+    this.faces.forEach(face => {
+      // OBJ format uses 1-based indices
+      const faceStr = face.map(index => index + 1).join(' ');
+      objStr += `f ${faceStr}\n`;
+    });
+
+    save(objStr, fileName);
```

Comment:
`save()` is a user-facing function, so if this method uses it, users could also technically use it, but they would have to know to pass an obj string through here. Maybe instead we could use the internal `downloadFile` method?

https://github.com/processing/p5.js/blob/37d3324b457b177319ed65468201f2806a66eff5/src/image/image.js#L227

Reply:
![image](https://github.com/processing/p5.js/assets/110971977/c0dbaccc-74a3-4f70-982d-acc517acb24e)
using `downloadFile` gives this error while `save()` seamlessly downloads the `.obj` file. As far as I know downlaodFile did not need to be updated for any particular extension like `save()` so it should have worked I guess?